typescript/apps/washboard-ui/src/app/components/providers-table.tsx

```diff
@@ -69,21 +75,12 @@ const columns = [
     id: 'health',
     header: 'Health',
     cell: (info) => {
-      const healthSummary: 'Running' | 'Pending' | 'Failed' = Object.values(info.getValue()).reduce(
-        (summary, currentStatus): 'Running' | 'Pending' | 'Failed' => {
-          // health status can be 'Running', 'Pending' or 'Failed'
-          if (summary === 'Failed') {
-            return summary;
-          } else if (summary === 'Pending') {
-            return currentStatus === 'Failed' ? 'Failed' : 'Pending';
-          } else if (currentStatus === 'Running') {
-            return currentStatus;
-          } else {
-            return 'Pending';
-          }
-        },
-        'Running',
-      ) as 'Running' | 'Pending' | 'Failed';
+      const healthSummary = Object.values(info.getValue()).reduce((summary, currentStatus) => {
+        if (summary === 'Failed' || currentStatus === 'Failed') return 'Failed';
+        if (summary === 'Pending' || currentStatus === 'Pending') return 'Pending';
+        if (summary === 'Running' && currentStatus === 'Running') return 'Running';
+        return 'Pending';
+      }, 'Running');
```

Comment:
Much cleaner. Love it!

I've never liked the names of these parameters but won't hold it back as is. If you think of better ones, feel free to change them!

Reply:
Well I hesitated to change it to an `Array.prototype.some` and `Array.prototype.every` function with the same logic, but I think it should be good enough. Maybe people who touch it next time will change it.